### ISSUE TITLE
Too many sessions?

### ISSUE THREAD

**Issue by sorin-davidoi** [tootsuite/mastodon#4058]
I am debugging my push notification PR and I and noticed that sessions seem to be created too often. The following is from my account at toot.cafe:

![screenshot_20170703_204956](https://user-images.githubusercontent.com/2109702/27804895-a8707088-6031-11e7-9439-b3fb9a60b558.png)

Each time I checked my phone during the day a new session was created. I did not use Incognito Mode and don't have any extensions installed. I notice the same behavior for my account at glitch.social.

If I understand the feature correctly, since I am logged in in only two places, there should only be two entries in that table (the irony is not lost on me - I made the original PR to introduce session activations).

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by nightpool** [comment]
I haven't seen this behavior, chrome on iOS, chrome on mac.

**Comment by sorin-davidoi** [comment]
I think this is why we keep getting "Subscription registered" on glitch.social. Since the subscriptions are attached to sessions registration, for each new registration a new push subscription is generated.

**Comment by beatrix-bitrot** [comment]
on glitch.social i have almost a dozen but they are not created this often... mosto recent one was created 2 days ago apparently

**Comment by beatrix-bitrot** [comment]
aha! I can confirm this now

 Chrome on Android	xxx.xxx.xxx.xxx 	1 hour ago

and I got the new push subscription to go with it

no idea why it would create a new session

**Comment by nightpool** [comment]
yeah nevermind this is happening to me chrome on ios too. seems to be related to how mobile chrome works.....

**Comment by sorin-davidoi** [comment]
Don't think so, as it also happens for Chrome on desktop in my case.

**Comment by Gargron** [comment]
Devise uses sessions (and I mean this in a cookie vs session sense, not our own layer of abstraction), sessions are impermanent. To do the "stay logged in for 2 weeks thing", it uses a remember_me cookie, which revives the session. This is why there are so many sessions, because they expire quickly. Our layer of abstraction should rely on the remember_me cookie.

**Comment by beatrix-bitrot** [comment]
@sorin-davidoi can it be reworked to use the cookie like garg says?

**Comment by sorin-davidoi** [comment]
I suppose it can, but that should be in a separate PR - not related to push notifications. I'm afraid I don't really know how to approach it though, would appreciate if someone more knowledgeable in this area could give a hand.

**Comment by beatrix-bitrot** [comment]
@sorin-davidoi that sounds reasonable to me, i guess that would be something we try to get into master sooner than later so that this can be unblocked?

@Gargron it sounds like you have something in mind for the layer of abstraction and remember_me cookie and stuff. is this something you can implement or recommend someone else we can tag in to get help from?

**Comment by nolanlawson** [comment]
@nightpool Chrome on iOS is basically a wrapper around Safari (WebKit): https://daringfireball.net/linked/2012/06/28/chrome-ios

**Comment by nightpool** [comment]
@nolanlawson yeah? confused about how that relates to what I said. I was seeing it on mobile, but not on my computer.

**Comment by nightpool** [comment]
also it's not *exactly* a wrapper around safari—it's an IOSWebView, which can have some slight quirks compared to standard ios safari.
